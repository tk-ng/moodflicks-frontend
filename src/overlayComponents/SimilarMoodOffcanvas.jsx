import "./AppearedOnOffcanvas.css";
import { MoodList } from "../moodComponents/MoodList";
import { Offcanvas, OffcanvasBody, OffcanvasHeader } from "reactstrap";

export const SimilarMoodOffcanvas = ({ canvas, toggle, moods }) => {
	return (
		<div>
			<Offcanvas
				direction="bottom"
				toggle={toggle}
				isOpen={canvas}
				scrollable={true}
				backdrop
			>
				<OffcanvasHeader toggle={toggle}>
					The following moods have similar vibes to this mood list:
				</OffcanvasHeader>
				<OffcanvasBody>
					<MoodList moods={moods} />
				</OffcanvasBody>
			</Offcanvas>
		</div>
	);
};
