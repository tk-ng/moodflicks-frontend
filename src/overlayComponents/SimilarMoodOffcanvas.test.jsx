import { render, screen } from "@testing-library/react";
import { SimilarMoodOffcanvas } from "./SimilarMoodOffcanvas";
import { MemoryRouter } from "react-router";

describe("Render the SimilarMoodOffcanvas component", function () {
	const toggle = vi.fn();
	const testMoods = [
		{
			movies: ["testid1", "testid2", "testid3", "testid4"],
			count: 4,
			mood: "testmood-1",
			id: "8",
			createdBy: "testuser2",
			totalVotes: "2",
		},
		{
			movies: ["testid5", "testid6", "testid7", "testid8", "testid9"],
			count: 5,
			mood: "testmood-2",
			id: "11",
			createdBy: "testuser1",
			totalVotes: "0",
		},
	];
	it("mounts without crashing", async function () {
		const { container, asFragment } = render(
			<SimilarMoodOffcanvas canvas={true} toggle={toggle} moods={[]} />
		);
		expect(container).toBeInTheDocument();
		expect(asFragment).toMatchSnapshot();
	});

	it("renders with moods", async function () {
		render(
			<MemoryRouter>
				<SimilarMoodOffcanvas canvas={true} toggle={toggle} moods={testMoods} />
			</MemoryRouter>
		);
		expect(screen.getByText("testmood-1")).toBeInTheDocument();
		expect(screen.getByText("testmood-2")).toBeInTheDocument();
	});

	// Without similar moods the "Similar Moods" button is not displayed. Tested in MoodDetail.test.jsx
});
