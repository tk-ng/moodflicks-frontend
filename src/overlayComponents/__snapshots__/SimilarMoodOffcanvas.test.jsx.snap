// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render the SimilarMoodOffcanvas component > mounts without crashing 1`] = `[Function]`;
